packages/front-end/components/Dimensions/DimensionForm.tsx

```diff
@@ -18,14 +18,16 @@ const DimensionForm: FC<{
   const { memberUsernameOptions } = useMembers();
   const {
     getDatasourceById,
-    datasources,
+    datasourcesFilteredByProject: filteredDatasources,
```

Comment:
This code has nothing to do with permissions.  It's just filtering the data sources based on the currently selected project in the left nav.  The problem is that we use this same form when editing existing dimensions.  If a dimension's existing data source is not part of the current project, when you go to edit the select box will be broken since the value won't be one of the options.